legend-engine-executionPlan-execution/src/main/java/org/finos/legend/engine/plan/execution/stores/StoreExecutableManager.java

```diff
@@ -0,0 +1,104 @@
+// Copyright 2023 Goldman Sachs
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//      http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+package org.finos.legend.engine.plan.execution.stores;
+
+import org.eclipse.collections.api.list.MutableList;
+import org.eclipse.collections.api.multimap.MutableMultimap;
+import org.eclipse.collections.impl.multimap.list.FastListMultimap;
+import org.finos.legend.engine.shared.core.operational.logs.LogInfo;
+import org.finos.legend.engine.shared.core.operational.logs.LoggingEventType;
+import org.slf4j.Logger;
+
+import javax.servlet.http.HttpSessionEvent;
+import javax.servlet.http.HttpSessionListener;
+
+
+/**
+ * This is a singleton that maintains the relationship between  http sessions and store executions.
+ * It can be used to operate on executions (for example to cancel running executions)
+ * The manager must be initialized as well as added as a HttpSessionListener
+ */
+public enum StoreExecutableManager implements HttpSessionListener
```

Comment:
Can we avoid the singleton here, and just let the sever create the one instance we need, and propagate it accordantly?  Or even further, do we need to this inside the Result class?  Cannot we do this management on the Server resource that will end serializing the result?

Reply:
The singleton is intended to ensure there is only one instance of the manager created. That is critical to ensure the state of running executions is tracked correctly.